enclave/core/sgx/hostcalls.c

```diff
@@ -3,18 +3,78 @@
 
 #include <openenclave/edger8r/enclave.h>
 #include <openenclave/enclave.h>
+#include <openenclave/internal/ecall_context.h>
+#include "td.h"
 
-// Function used by oeedger8r for allocating ocall buffers. This function can be
-// optimized by allocating a buffer for making ocalls and pass it in to the
-// ecall and making it available for use here.
+/**
+ * Validate and fetch this thread's ecall context.
+ */
+static oe_ecall_context_t* _get_ecall_context()
+{
+    td_t* td = oe_get_td();
```

Comment:
Just out of curiosity, have OE checked the td is initialized anywhere? It seems OE assumes the `oe_get_id` always get initialized data.